olive/evaluator/olive_evaluator.py

```diff
@@ -141,10 +142,12 @@ def _evaluate_custom(
     ) -> MetricResult:
         raw_res = None
         if metric.user_config.evaluate_func:
+            data_dir = metric.data_config.load_dataset_params.get("data_dir") if metric.data_config else None
+            batch_size = metric.data_config.dataloader_params.get("batch_size", 1) if metric.data_config else 1
             raw_res = eval_func(
```

Comment:
since, I already made a PR which is able to automatically detect data resources, I think it might be simpler here to pass any values the evaluate func needs as `evaluate_kwargs`. This way we don't need a data config for a custom evaluator func.